sharding/nodesCoordinator/nodesCoordinatorRegistryFactory.go

```diff
@@ -3,78 +3,75 @@ package nodesCoordinator
 import (
 	"encoding/json"
 
-	"github.com/ElrondNetwork/elrond-go-core/core/atomic"
 	"github.com/ElrondNetwork/elrond-go-core/core/check"
 	"github.com/ElrondNetwork/elrond-go-core/marshal"
 )
 
 type nodesCoordinatorRegistryFactory struct {
-	stakingV4EnableEpoch uint32
-	flagStakingV4        atomic.Flag
 	marshaller           marshal.Marshalizer
+	stakingV4EnableEpoch uint32
 }
 
 // NewNodesCoordinatorRegistryFactory creates a nodes coordinator registry factory which will create a
 // NodesCoordinatorRegistryHandler from a buffer depending on the epoch
 func NewNodesCoordinatorRegistryFactory(
 	marshaller marshal.Marshalizer,
-	notifier EpochNotifier,
 	stakingV4EnableEpoch uint32,
 ) (*nodesCoordinatorRegistryFactory, error) {
 	if check.IfNil(marshaller) {
 		return nil, ErrNilMarshalizer
 	}
-	if check.IfNil(notifier) {
-		return nil, ErrNilEpochNotifier
-	}
 
-	log.Debug("nodesCoordinatorRegistryFactory: staking v4 enable epoch", "epoch", stakingV4EnableEpoch)
-
-	ncf := &nodesCoordinatorRegistryFactory{
+	return &nodesCoordinatorRegistryFactory{
 		marshaller:           marshaller,
 		stakingV4EnableEpoch: stakingV4EnableEpoch,
-	}
-	notifier.RegisterNotifyHandler(ncf)
-	return ncf, nil
+	}, nil
 }
 
 // CreateNodesCoordinatorRegistry creates a NodesCoordinatorRegistryHandler depending on the buffer. Old version uses
 // NodesCoordinatorRegistry with a json marshaller; while the new version(from staking v4) uses NodesCoordinatorRegistryWithAuction
 // with proto marshaller
 func (ncf *nodesCoordinatorRegistryFactory) CreateNodesCoordinatorRegistry(buff []byte) (NodesCoordinatorRegistryHandler, error) {
-	if ncf.flagStakingV4.IsSet() {
-		return ncf.createRegistryWithAuction(buff)
+	registry, err := ncf.createRegistryWithAuction(buff)
+	if err == nil {
```

Comment:
Agree, but this change is actually a fix; since flag activation wouldn't work.
I've discovered a `chicken-egg` problem with this flag in case of node restart(also raised this problem to @iulianpascalau and @AdoAdoAdo) . After node restart, calling `getLastBootstrapData` was done with some dummy values and `nodesCoordinatorRegistryFactory` wouldn't have the flag correctly set causing unmarshal error.